### PR TITLE
[POC] Add a demo contract of gas less paymaster

### PR DESCRIPTION
**Pull request by magicalne**

### CODE REVIEW

File: test/gasless_paymaster.test.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          maxFeePerGas: 1,
+          maxPriorityFeePerGas: 1,
+          paymasterAndData: hexConcat([paymaster.address, "0x1234"])
+      }
+      
+      // Send tx with a valid user.
+      entryPoint.connect(whitelistUser)
+
+      // Should be OK
+      const res = await entryPoint.handleOp(userOp, {gasLimit: 100000, gasPrice: 0}) // FIXME: How to set 0 gas price?
```
Comment (magicalne) [inline test/gasless_paymaster.test.ts]: Currently, I cannot make it work on hardhat.  @RetricSu any thoughts?

Comment (RetricSu) [reply test/gasless_paymaster.test.ts]: it seems hardhat network block the 0 gas price. I think we should turn on `gas price = 0` in `hardhat.config.ts` https://hardhat.org/hardhat-network/docs/reference#gasprice

Comment (magicalne) [reply test/gasless_paymaster.test.ts]: It works with config:
```ts
    hardhat: {
      gasPrice: 0,
      gas: "auto",
      gasMultiplier: 0,
      initialBaseFeePerGas: 0,
    }
```

File: test/gasless_paymaster.test.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+          callData,
+          callGasLimit: 10000,
+          verificationGasLimit: 1000000,
+          maxFeePerGas: 1,
+          maxPriorityFeePerGas: 1,
+          paymasterAndData: hexConcat([paymaster.address, "0x1234"])
+      }
+      
+      // Send tx with a valid user.
+      await entryPoint.connect(whitelistUser).handleOp(userOp, {gasLimit: 100000, gasPrice: 0})
```
Comment (RetricSu) [inline test/gasless_paymaster.test.ts:65]: shall we check some results here:

1. contract state after execution: `const sum = await DummyContract.sum()` expect `sum` to be equal to `2`;
2. paymaster's deposit value is reduced just the amount of gas fee: `gas fee == paymaster deposit value reduction`

**Review by RetricSu — APPROVED** [review]

**Pull request merged**